sklearn/model_inspection/partial_dependence.py

```diff
@@ -0,0 +1,356 @@
+                )
+            elif response_method == 'predict_proba':
+                raise ValueError('The estimator has no predict_proba method.')
+            else:
+                raise ValueError(
+                    'The estimator has no decision_function method.')
+
+    for new_values in grid:
+        X_eval = X.copy()
+        for i, variable in enumerate(features):
+            X_eval[:, variable] = new_values[i]
+
+        try:
+            predictions = prediction_method(X_eval)
+        except NotFittedError:
+            raise ValueError('est parameter must be a fitted estimator')
+
+        # Note: predictions is of shape
+        # (n_points,) for non-multioutput regressors
+        # (n_points, n_tasks) for multioutput regressors
+        # (n_points, 1) for the regressors in cross_decomposition (I think)
+        # (n_points, 2)  for binary classifaction
+        # (n_points, n_classes) for multiclass classification
+
+        # average over samples
+        averaged_predictions.append(np.mean(predictions, axis=0))
+
+    # reshape to (n_targets, n_points) where n_targets is:
+    # - 1 for non-multioutput regression and binary classification (shape is
+    #   already correct in those cases)
+    # - n_tasks for multi-output regression
+    # - n_classes for multiclass classification.
+    averaged_predictions = np.array(averaged_predictions).T
+    if is_regressor(est) and averaged_predictions.ndim == 1:
+        # non-multioutput regression, shape is (n_points,)
+        averaged_predictions = averaged_predictions.reshape(1, -1)
+    elif is_classifier(est) and averaged_predictions.shape[0] == 2:
+        # Binary classification, shape is (2, n_points).
+        # we output the effect of **positive** class
+        averaged_predictions = averaged_predictions[1]
+        averaged_predictions = averaged_predictions.reshape(1, -1)
+
+    return averaged_predictions
+
+
+def partial_dependence(est, features, X, response_method='auto',
+                       percentiles=(0.05, 0.95), grid_resolution=100,
+                       method='auto'):
+    """Partial dependence of ``features``.
+
+    Partial dependence of a feature (or a set of features) corresponds to
+    the average response of an estimator for each possible value of the
+    feature.
+
+    Read more in the :ref:`User Guide <partial_dependence>`.
+
+    Parameters
+    ----------
+    est : BaseEstimator
+        A fitted classification or regression model. Multioutput-multiclass
+        classifiers are not supported.
+    features : list or array-like of int
+        The target features for which the partial dependency should be
+        computed.
+    X : array-like, shape=(n_samples, n_features)
+        ``X`` is used both to generate a grid of values for the
+        ``features``, and to compute the averaged predictions when
+        method is 'brute'.
+    response_method : 'auto', 'predict_proba' or 'decision_function', \
```

Comment:
Here, we might want to think twice about the name of the parameters. In `cross_val_predict`, we call it simply `method` while in the `StackingClassifier`, we used `method_estimators`. I don't know if it would make sense to have a consistent name since that in each estimator, the role of the parameter is quite the same. WDYT?

Reply:
We could also think if it makes sense to have a shared utils for the `'auto'` decision across those estimators.

Reply:
There's already a parameter called `method`.

IMO `response_method` makes sense here, in the context of PDPs.

I don't think having a shared utils for 'auto' is relevant here since the way we choose the method is very specific to PDPs.